### PR TITLE
fix non-affine case and add test for unbiased Nitsche

### PR DESCRIPTION
**Pull request by SarahRo**

### CODE REVIEW

File: cpp/utils.h [hunk 1]
```diff
@@ -116,4 +116,29 @@ void evaluate_basis_functions(const dolfinx::fem::FunctionSpace& V,
...
 
+/// Compute physical normal
+/// @param[in] n_ref facet normal on reference element
+/// @param[in] K inverse Jacobian
+/// @param[out] n_phys facet normal on physical element
+void compute_normal(const xt::xtensor<double, 1>& n_ref,
+                    const xt::xtensor<double, 2>& K,
+                    xt::xarray<double>& n_phys);
+
+/// Compute jacobians on a given facet
```
Comment (jorgensd) [inline cpp/utils.h:127]: Could you specify that you compute the Jacobian from physical cell -> reference cell and physical facet -> reference facet. Currently I thought the name indicated that you computed `J_f`

File: cpp/utils.h [hunk 2]
```diff
@@ -116,4 +116,29 @@ void evaluate_basis_functions(const dolfinx::fem::FunctionSpace& V,
...
+/// @param[in] q - index of quadrature points
+/// @param[in] dphi - derivatives of coordinate basis tabulated for quardrature
+/// points
+/// @param[in] coords - the coordinates of the facet
+/// @param[in] J_f - the Jacobian between reference facet and reference cell
+/// @param[out] J - Jacboian between reference cell and physical cell
+/// @param[out] K - inverse of J
+/// @param[out] J_tot - J_f*J
```
Comment (jorgensd) [inline cpp/utils.h:135]:
```suggestion
/// @param[in,out] J - Jacboian between reference cell and physical cell
/// @param[in,out] K - inverse of J
/// @param[in,out] J_tot - J_f*J
```

File: cpp/utils.h [hunk 3]
```diff
@@ -116,4 +116,29 @@ void evaluate_basis_functions(const dolfinx::fem::FunctionSpace& V,
                               const xtl::span<const std::int32_t>& cells,
                               xt::xtensor<double, 3>& basis_values);
 
+/// Compute physical normal
+/// @param[in] n_ref facet normal on reference element
+/// @param[in] K inverse Jacobian
+/// @param[out] n_phys facet normal on physical element
```
Comment (jorgensd) [inline cpp/utils.h:122]:
```suggestion
/// @param[in,out] n_phys facet normal on physical element
```

File: python/tests/test_unbiased.py [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+    # fem functions
+
+    def _u1(x):
+        values = np.zeros((gdim, x.shape[1]))
+        for i in range(tdim):
+            for j in range(x.shape[1]):
+                if i == tdim - 1:
+                    values[i, j] = np.sin(x[i, j] + gap)
+                else:
+                    values[i, j] = np.sin(x[i, j])
```
Comment (jorgensd) [inline python/tests/test_unbiased.py:348]:
```suggestion
            values[i] = np.sin(x[i] + gap) if i == tdim - 1 else  np.sin(x[i])
```

File: python/tests/test_unbiased.py [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+        return values
+
+    u1 = _fem.Function(V_cuas)
+    v1 = ufl.TestFunction(V_cuas)
+
+    u1.interpolate(_u0, cells[0])
+    u1.interpolate(_u1, cells[1])
+
```
Comment (jorgensd) [inline python/tests/test_unbiased.py:356]:
```suggestion
u1.x.scatter_forward()
```
To make sure this test would work in parallel (even if we will never run it in parallel).

File: cpp/utils.h (hunk 1 above)
Comment (SarahRo) [inline cpp/utils.h:127]:
```suggestion
/// Compute the following jacobians on a given facet:
/// J: physical cell -> reference cell (and its inverse)
/// J_tot: physical facet -> reference facet
```

**Pull request merged**